lib/js/src/manager/permission/PermissionManagerBase.js

```diff
@@ -0,0 +1,376 @@
+* disclaimer in the documentation and/or other materials provided with the
+* distribution.
+*
+* Neither the name of the Livio Inc. nor the names of its contributors
+* may be used to endorse or promote products derived from this software
+* without specific prior written permission.
+*
+* THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+* AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+* IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+* ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+* LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+* CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+* SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+* INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+* CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+* ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+* POSSIBILITY OF SUCH DAMAGE.
+*/
+
+import { BaseSubManager } from '../BaseSubManager.js';
+import { PermissionFilter } from './PermissionFilter.js';
+import { PermissionGroupType } from './enums/PermissionGroupType.js';
+import { PermissionGroupStatus } from './enums/PermissionGroupStatus.js';
+import { PermissionStatus } from './PermissionStatus.js';
+import { FunctionID } from '../../rpc/enums/FunctionID.js';
+import { PredefinedWindows } from '../../rpc/enums/PredefinedWindows.js';
+
+class PermissionManagerBase extends BaseSubManager {
+    constructor (lifecycleManager) {
+        super(lifecycleManager);
+        this._currentHmiLevel = null;
+        this._currentPermissionItems = {}; // a key:value store of FunctionId permissions
+        this._encryptionRequiredRpcs = []; // an array of FunctionIDs which require encryption
+        this._filters = [];
+
+        this._onHMIStatusListener = (notification) => {
+            if (notification.getWindowID() !== null && notification.getWindowID() !== PredefinedWindows.DEFAULT_WINDOW) {
+                return;
+            }
+            const previousHmiLevel = this._currentHmiLevel;
+            this._currentHmiLevel = notification.getHmiLevel();
+            this._checkState();
+            this._notifyListeners(this._currentPermissionItems, previousHmiLevel, this._currentPermissionItems, this._currentHmiLevel);
+        };
+        this._lifecycleManager.addRpcListener(FunctionID.OnHMIStatus, this._onHMIStatusListener);
+
+        this._onPermissionsChangeListener = (notification) => {
+            const permissionItems = notification.getPermissionItem();
+            const previousPermissionItems = JSON.parse(JSON.stringify(this._currentPermissionItems));
+            const requireEncryptionAppLevel = notification.getRequireEncryption();
+            this._encryptionRequiredRpcs.splice(0, this._encryptionRequiredRpcs.length);
```

Comment:
maybe state that this line clears the array out?

Reply:
Resolved in https://github.com/smartdevicelink/sdl_javascript_suite/pull/116/commits/a5f638b42612168418303acc8d49b14740469b00